src/libtrx/game/items/walkable.c

```diff
@@ -0,0 +1,167 @@
+        pos.x, pos.y, pos.z, sector->floor.height, sector->ceiling.height);
+    WALKABLE *node = GameBuf_Alloc(sizeof(WALKABLE), GBUF_WALKABLES);
+    node->item_num = item_num;
+    node->pos = pos;
+    node->next = NULL;
+    M_InsertSorted(&sector->walkable, node);
+}
+
+void Walkable_Add(const int16_t item_num)
+{
+    ITEM *item = Item_Get(item_num);
+    if (item->object_id == O_DRAWBRIDGE) {
+        LOG_DEBUG(
+            "item %d OG pos: %d %d %d", item_num, item->pos.x, item->pos.y,
+            item->pos.z);
+        VECTOR *drawbridge_pos = Vector_Create(sizeof(XYZ_32));
+        Drawbridge_GetSectorPositions(item, drawbridge_pos);
+        for (int32_t i = 0; i < drawbridge_pos->count; i++) {
+            M_Add(
+                item_num, item, *(const XYZ_32 *)Vector_Get(drawbridge_pos, i));
+        }
+        Vector_Free(drawbridge_pos);
+        drawbridge_pos = nullptr;
+    } else if (
+        item->object_id == O_TRAPDOOR_TYPE_1
+        || item->object_id == O_TRAPDOOR_TYPE_2
+        || item->object_id == O_TRAPDOOR_TYPE_3) {
+        LOG_DEBUG(
+            "Trapdoor item %d OG pos: %d %d %d", item_num, item->pos.x,
+            item->pos.y, item->pos.z);
+        VECTOR *trapdoor_pos = Vector_Create(sizeof(XYZ_32));
+        Trapdoor_GetSectorPositions(item, trapdoor_pos);
+        for (int32_t i = 0; i < trapdoor_pos->count; i++) {
+            M_Add(item_num, item, *(const XYZ_32 *)Vector_Get(trapdoor_pos, i));
+        }
+        Vector_Free(trapdoor_pos);
+        trapdoor_pos = nullptr;
+    } else {
+        LOG_DEBUG(
+            "Generic add item %d OG pos: %d %d %d", item_num, item->pos.x,
+            item->pos.y, item->pos.z);
+        M_Add(item_num, item, item->pos);
+    }
+}
+
+void Walkable_Remove(const int16_t item_num)
+{
+    const ITEM *const item = Item_Get(item_num);
+    SECTOR *const sector = M_GetItemPitSector(item, item->pos);
+
+    WALKABLE **walkables = &sector->walkable;
+    while (*walkables) {
+        WALKABLE *node = *walkables;
+        if (node->item_num == item_num) {
+            *walkables = node->next;
+            // LOG_DEBUG("Unlink item_num: %d", item_num);
+            return;
+        }
+        walkables = &node->next;
+    }
+}
+
+void Walkable_Reposition(int16_t item_num, const XYZ_32 old_pos)
+{
+    ITEM *item = Item_Get(item_num);
+    SECTOR *old_sector = M_GetItemPitSector(item, old_pos);
+    LOG_DEBUG(
+        "TRY reposition %d from %d %d %d to %d %d %d", item_num, old_pos.x,
+        old_pos.y, old_pos.z, item->pos.x, item->pos.y, item->pos.z);
+
+    // Unlink the walkable from the old position.
+    WALKABLE **walkables = &old_sector->walkable;
+    WALKABLE *node = NULL;
+    while (*walkables) {
+        if ((*walkables)->item_num == item_num) {
+            node = *walkables;
+            *walkables = node->next;
+            break;
+        }
+        walkables = &(*walkables)->next;
+    }
+    if (!node) {
```

Comment:
Rather than `(!node)` we do null checks very explicitly `(node == nullptr)`. (Here and other places)